webapp/src/reducers.ts

```diff
@@ -778,6 +791,49 @@ const recentlyJoinedUsers = (state: recentlyJoinedUsersState = {}, action: conne
     }
 };
 
+type IncomingCallAction = {
+    type: string;
+    data: {
+        callID: string;
+        hostID: string;
+        startAt: number;
+        ring: boolean;
+        type: ChannelType;
+    },
+};
+
+const incomingCalls = (state: IncomingCallNotification[] = [], action: IncomingCallAction) => {
+    switch (action.type) {
+    case ADD_INCOMING_CALL:
+        return [...state, {...action.data}];
+    case REMOVE_INCOMING_CALL:
+        return state.filter((ic) => ic.callID !== action.data.callID);
+    case CALL_HAS_ENDED:
+        return state.filter((ic) => ic.callID !== action.data.callID);
+    default:
+        return state;
+    }
+};
+
+type HaveRangForCallsAction = {
```

Comment:
Shouldn't this be "HaveRung" grammatically speaking?
Honestly I'd be fine with a simpler "DidRing" but clearly not a big deal.